base/asyncmap.jl

```diff
@@ -145,4 +145,27 @@ function next(itr::StreamMapIterator, state::StreamMapState)
     return (r, state)
 end
 
-iteratorsize(::Type{StreamMapIterator}) = SizeUnknown()
+iteratorsize(::Type{StreamGenerator}) = SizeUnknown()
+
+
+"""
+    asyncgenerate(f, c...) -> iterator
+
+Apply `@async f` to each element of `c`.
+
+For multiple collection arguments, apply f elementwise.
+
+Results are returned in order as they become available.
+"""
+asyncgenerate(f, c...) = StreamGenerator(f, c...)
```

Comment:
> I think I agree that the thing that is currently called StreamGenerator should probably be called AsyncGenerator.
> I could roll these together so that the type currently called AsyncGenerator goes away? what do you think?

How would that work?